docs/dev/compliance.md

```diff
@@ -0,0 +1,120 @@
+
+#### Legacy external require location
+
+It appears that something - as in some other software package - is looking to require certain libs from paths like `inspec/lib/bundles/inspec-compliance`.  There are several stub files at that location. TODO - determine what software is doing that and either discourage such behavior or at least point them at the right location.
+
+### Important files
+
+#### lib/cli.rb
+
+This file defines the Thor subcommand CLI UX.  Look here for all commands and options. Most command implementation is right here inline, which makes this file large and noisy. Some implementation pertaining to interaction with the actual Automate service is pushed out to a class `InspecPlugins::Compliance::API`; class methods are used to perform actions.
+
+#### lib/configuration.rb
+
+A fairly straightforward data container with JSON marshalling. Used to store CLI options as well as to write credentials to disk.
+
+#### lib/api.rb
+
+This is the wrapper around the Automate API. The approach is to present class methods which take as an argument a stateful config object.
+The code is brittle, being highly conditionalized for products that have been sunsetted, such as Compliance and the soon to sunsetted Automate1. In some cases there are conditionals for versions that have been sunsetted as well.
+
+Actual HTTP communication is handled by `InspecPlugins::Compliance::HTTP`, again using class methods.
+
+#### lib/http.rb
+
+This is probably unneccesary. It is a wrapper around Net:HTTP. Instead, we should probably be using a REST API wrapper or something similar.
+
+#### lib/support.rb
+
+Tries to map versions to feature support for the API. Does not appear to be relied on very much. Duplicate functionality may be in lib/configuration.rb .
+
+## Subcommands
+
+There are several other minor commands not listed here - see `lib/cli.rb` for a complete listing.
+
+### login
+
+Saves a credentials file locally. Future invocations of `inspec compliance` use the credentials file to authenticate.
+
+`be inspec compliance login --user=admin --token='1234567890asdfghjkl' --insecure https://chef-automate.test`
+
+Here are the results of running login, from `.inspec/compliance/config.json`:
+
+```json
+{
+	"automate": {
+		"ent": "automate",
+		"token_type": "dctoken"
+	},
+	"server": "https://chef-automate.test/api/v0",
+	"user": "admin",
+	"owner": "admin",
+	"insecure": true,
+	"server_type": "automate2",
+	"token": "1234567890asdfghjkl",
+	"version": "0"
+}
+```
+
+#### login code trace
+
+Thor code in `lib/cli.rb` stores the passed-in server URL in the config object, then passes everything to `InspecPlugins::Compliance::API.login`. That class method is actually defined in `lib/api/login.rb` and is quite large because it immediately conditionalizes on supporting three products - Compliance, A1, and A2.
```

Comment:
```suggestion
The Thor code in `lib/cli.rb` stores the passed-in server URL in the config object, and then passes everything to `InspecPlugins::Compliance::API.login`. That class method is actually defined in `lib/api/login.rb` and is quite large because it immediately conditionalizes on supporting three products - Compliance, A1, and A2.
```